tools/workflow/cleaner/lambda/clean.go

```diff
@@ -15,48 +16,120 @@ const Type = "lambda"
-		ListFunctionsOutput, err := lambdaclient.ListFunctions(ListFunctionsInput)
+		listFunctionsInput := &lambda.ListFunctionsInput{Marker: nextToken}
+		listFunctionsOutput, err := lambdaClient.ListFunctions(listFunctionsInput)
 		if err != nil {
-			return fmt.Errorf("unable to retrieve APIs: %w", err)
+			return fmt.Errorf("unable to retrieve functions: %w", err)
 		}
-		for _, lf := range ListFunctionsOutput.Functions {
+
+		for _, lf := range listFunctionsOutput.Functions {
 			doesNameMatch, err := shouldDelete(lf)
 			if err != nil {
 				return fmt.Errorf("error during pattern match: %w", err)
 			}
 			created, err := time.Parse("2006-01-02T15:04:05Z0700", *lf.LastModified)
 			if err != nil {
-				return fmt.Errorf("error parting last modified time: %w", err)
+				return fmt.Errorf("error parsing last modified time: %w", err)
 			}
 			if expirationDate.After(created) && doesNameMatch {
 				logger.Printf("Try to delete function %s created-at %s", *lf.FunctionArn, *lf.LastModified)
 				deleteFunctionIDs = append(deleteFunctionIDs, lf.FunctionArn)
 			}
 		}
-		if ListFunctionsOutput.NextMarker == nil {
+		if listFunctionsOutput.NextMarker == nil {
 			break
 		}
-		nextToken = ListFunctionsOutput.NextMarker
+		nextToken = listFunctionsOutput.NextMarker
 	}
 	if len(deleteFunctionIDs) < 1 {
 		logger.Printf("No Lambda functions to delete")
 		return nil
 	}
-
 	for _, id := range deleteFunctionIDs {
 		terminateFunctionInput := &lambda.DeleteFunctionInput{FunctionName: id}
-		if _, err := lambdaclient.DeleteFunction(terminateFunctionInput); err != nil {
+		if _, err := lambdaClient.DeleteFunction(terminateFunctionInput); err != nil {
 			return fmt.Errorf("unable to delete function: %w", err)
 		}
+		logger.Printf("Deleted %d Lambda functions", len(deleteFunctionIDs))
 	}
+
+	return nil
+}
+
+func cleanLayers(sess *session.Session, expirationDate time.Time) error {
+	logger.Printf("Begin to clean Lambda layers")
+	lambdaClient := lambda.New(sess)
+	var deleteLayerVersions []Layer
+	var nextToken *string
+
+	for {
+		listLayerVersionsInput := &lambda.ListLayersInput{
+			Marker: nextToken,
+		}
+
+		// Retrieve layer versions from Lambda service
+		listLayerVersionsOutput, err := lambdaClient.ListLayers(listLayerVersionsInput)
+		if err != nil {
+			return fmt.Errorf("unable to retrieve layer versions: %w", err)
+		}
+
+		// Loop through retrieved layer versions
+		for _, layer := range listLayerVersionsOutput.Layers {
+			layerARN := shouldDeleteLayer(layer)
+			if layerARN != nil {
+				created, err := time.Parse("2006-01-02T15:04:05Z0700", *layer.LatestMatchingVersion.CreatedDate)
+				if err != nil {
+					return fmt.Errorf("error parsing layer created time: %w", err)
+				}
+				if expirationDate.After(created) {
+					logger.Printf("Try to delete layer version %s created-at %s", *layer.LayerArn, created)
+					deleteLayerVersions = append(deleteLayerVersions, Layer{layerARN, layer.LatestMatchingVersion.Version})
+				}
+			}
```

Comment:
```suggestion
			if(shouldDeleteLayer(layer))
			{
			  deleteLayerVersions = append(deleteLayerVersions, Layer{layerARN, layer.LatestMatchingVersion.Version})
			}
```

Reply:
We would like to check the created and delete them on timed basis, because we run soak tests for few hours, the tests would if the layer is missing in between, we also use the same layers to release during release workflow.

Reply:
I'm sorry. I'm unsure what that has to do with my suggestion above.

Reply:
sorry, disregard this, got your comment after refresh.